src/analyzer/analyzerrhythm.cpp

```diff
@@ -0,0 +1,268 @@
+    config.stepSize = stepSize;
+    config.frameLength = windowSize;
+    config.dbRise = 3;
+    config.adaptiveWhitening = 0;
+    config.whiteningRelaxCoeff = -1;
+    config.whiteningFloor = -1;
+    return config;
+}
+
+} // namespace
+
+
+AnalyzerRhythm::AnalyzerRhythm(UserSettingsPointer pConfig)
+        : m_iSampleRate(0),
+          m_iTotalSamples(0),
+          m_iMaxSamplesToProcess(0),
+          m_iCurrentSample(0),
+          m_iMinBpm(0),
+          m_iMaxBpm(9999) {
+}
+
+bool AnalyzerRhythm::initialize(TrackPointer pTrack, int sampleRate, int totalSamples) {
+    if (totalSamples == 0 or !shouldAnalyze(pTrack)) {
+        return false;
+    }
+
+    m_iSampleRate = sampleRate;
+    m_iTotalSamples = totalSamples;
+    m_iMaxSamplesToProcess = m_iTotalSamples;
+    m_iCurrentSample = 0;
+
+    m_stepSize = m_iSampleRate * kStepSecs;
+    m_windowSize = MathUtilities::nextPowerOfTwo(m_iSampleRate / kMaximumBinSizeHz);
+    m_pDetectionFunction = std::make_unique<DetectionFunction>(
+            makeDetectionFunctionConfig(m_stepSize, m_windowSize));
+    // decimation factor aims at resampling to c. 3KHz; must be power of 2
+    int factor = MathUtilities::nextPowerOfTwo(m_iSampleRate / 3000);
+    m_downbeat = std::make_unique<DownBeat>(
+            m_iSampleRate, factor, m_stepSize);
+
+    qDebug() << "input sample rate is " << m_iSampleRate << ", step size is " << m_stepSize;
+
+    m_processor.initialize(
+            m_windowSize, m_stepSize, [this](double* pWindow, size_t) {
+                m_detectionResults.push_back(
+                        m_pDetectionFunction->processTimeDomain(pWindow));
+                m_downbeat->pushAudioBlock(reinterpret_cast<float*>(pWindow));
+                return true;
+            });
+    
+    return true;
+}
+
+bool AnalyzerRhythm::shouldAnalyze(TrackPointer pTrack) const {
+    bool bpmLock = pTrack->isBpmLocked();
+    if (bpmLock) {
+        qDebug() << "Track is BpmLocked: Beat calculation will not start";
+        return true;
+    }
+    mixxx::BeatsPointer pBeats = pTrack->getBeats();
+    if (!pBeats) {
+        return true;
+    }
+    else if (!mixxx::Bpm::isValidValue(pBeats->getBpm())) {
+        qDebug() << "Re-analyzing track with invalid BPM despite preference settings.";
+        return true;
+    } else {
+        qDebug() << "Track already has beats, and won't re-analyze";
+        return false;
+    }
+    return true;
+}
+
+bool AnalyzerRhythm::processSamples(const CSAMPLE *pIn, const int iLen) {
+
+    m_iCurrentSample += iLen;
+    if (m_iCurrentSample > m_iMaxSamplesToProcess) {
+        return true; // silently ignore all remaining samples
+    }
+    return m_processor.processStereoSamples(pIn, iLen);;
+}
+
+void AnalyzerRhythm::cleanup() {
+    m_resultBeats.clear();
+    m_detectionResults.clear();
+    m_pDetectionFunction.reset();
+}
+
+std::vector<double> AnalyzerRhythm::computeBeats() {
+    std::vector<std::vector<double>> allBeats(kDfTypes);
```

Comment:
Please write this as: 
auto allBeats = std::vector<std::vector<double>>(kDfTypes);
else it looks like a function call.